components/percona_telemetry/component.cc

```diff
@@ -0,0 +1,97 @@
+#include <mysql/components/component_implementation.h>
+#include <mysql/components/my_service.h>
+#include <mysql/components/services/component_sys_var_service.h>
+#include <mysql/components/services/log_builtins.h>
+#include <mysql/components/services/mysql_command_consumer.h>
+#include <mysql/components/services/mysql_command_services.h>
+#include <mysql/components/services/mysql_current_thread_reader.h>
+#include <mysql/components/services/security_context.h>
+
+#include "common.h"
+#include "percona_telemetry_component.h"
+
+REQUIRES_SERVICE_PLACEHOLDER_AS(mysql_thd_security_context,
+                                thd_security_context_srv);
+REQUIRES_SERVICE_PLACEHOLDER_AS(mysql_command_thread, command_thread_srv);
+REQUIRES_SERVICE_PLACEHOLDER_AS(mysql_command_factory, command_factory_srv);
+REQUIRES_SERVICE_PLACEHOLDER_AS(mysql_command_options, command_options_srv);
+REQUIRES_SERVICE_PLACEHOLDER_AS(mysql_command_query, command_query_srv);
+REQUIRES_SERVICE_PLACEHOLDER_AS(mysql_command_query_result,
+                                command_query_result_srv);
+REQUIRES_SERVICE_PLACEHOLDER_AS(mysql_command_field_info,
+                                command_field_info_srv);
+REQUIRES_SERVICE_PLACEHOLDER_AS(mysql_command_error_info,
+                                command_error_info_srv);
+REQUIRES_SERVICE_PLACEHOLDER_AS(registry, registry_srv);
+REQUIRES_SERVICE_PLACEHOLDER_AS(log_builtins, log_builtins_srv);
+REQUIRES_SERVICE_PLACEHOLDER_AS(log_builtins_string, log_builtins_string_srv);
+REQUIRES_SERVICE_PLACEHOLDER_AS(component_sys_variable_register,
+                                component_sys_variable_register_srv);
+REQUIRES_SERVICE_PLACEHOLDER_AS(component_sys_variable_unregister,
+                                component_sys_variable_unregister_srv);
+
+static std::unique_ptr<PerconaTelemetryComponent> percona_telemetry_component;
```

Comment:
what is the advantage?

Reply:
I'm still not fully convinced, but let's do it in modern way as you said :)

1. Symbols in unnamed namespaces are exported from the library anyway, but their names are randomly mangled. This can cause the linker to do its job longer.
2. Maybe I'm old-school, but when I see `static`, I exactly know, what it means in all possible contexts. Having an unnamed namespace it is not instantly clear what the function scope is.
3. If there is only one static variable/function, wrapping it with an anonymous namespace seems to be too verbose. Using `static` in such a case seems to be violating rules: we should use one or another way of limiting the scope, not mixing to avoid mess.
4. Looks like C++ committee undepreciated use of `static`